app/controllers/katello/api/v2/host_collections_controller.rb

```diff
@@ -138,30 +153,33 @@ def copy
     private
 
     def find_host_collection
-      @organization = @system.organization if @system
       @organization = @activation_key.organization if @activation_key
-      @host_collection = HostCollection.where(:id => params[:id]).first
-      fail HttpErrors::NotFound, _("Couldn't find host collection '%s'") % params[:id] if @host_collection.nil?
+      id = params[:host_collection_id] || params[:id]
+      @host_collection = HostCollection.where(:id => id).first
+      fail HttpErrors::NotFound, _("Couldn't find host collection '%s'") % id if @host_collection.nil?
     end
 
     def host_collection_params
-      attrs = [:name, :description, :max_content_hosts, :unlimited_content_hosts, { :system_ids => [] }]
+      attrs = [:name, :description, :max_hosts, :unlimited_hosts, { :host_ids => [] }]
       params.fetch(:host_collection).permit(*attrs)
     end
 
-    def host_collection_params_with_system_uuids
+    def host_collection_params_with_host_ids
       result = host_collection_params
-      if params['system_uuids']
-        systems_from_uuid = System.uuids_to_ids(params['system_uuids'])
-        result['system_ids'] = result['system_ids'] ?  result['system_ids'] + systems_from_uuid : systems_from_uuid
-      end
-      result[:max_content_hosts] = nil if params[:unlimited_content_hosts]
+      result[:max_hosts] = nil if params[:unlimited_hosts]
       result
     end
 
-    def find_system
-      @system = System.find_by_uuid!(params[:system_id]) if params[:system_id]
-      @organization = @system.organization if @system
+    def find_host
+      @host = ::Host::Managed.find(params[:host_id]) if params[:host_id]
+
+      # This is necessary until the content host UI is updated to use host id instead of system UUID.
+      if params[:system_id]
+        @system = System.find_by_uuid!(params[:system_id]) if params[:system_id]
+        @host = @system.foreman_host
```

Comment:
See the comment above the if statement.  That will go away once the host and content host UI is merged.